go-controller/pkg/metrics/metrics.go

```diff
@@ -144,6 +170,158 @@ func coverageShowMetricsUpdater(component string) {
+			err = fmt.Errorf("recovering from a panic while parsing the stopwatch/show output for %s: %v", component, r)
+		}
+	}()
+
+	switch component {
+	case ovnController:
+		stdout, stderr, err = util.RunOVNControllerAppCtl("stopwatch/show")
+	case ovnNorthd:
+		stdout, stderr, err = util.RunOVNNorthAppCtl("stopwatch/show")
+	default:
+		return nil, fmt.Errorf("unknown component %s for stopwatch/show", component)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get stopwatch/show output for %s (stderr: %s): %w", component, stderr, err)
+	}
+
+	parsedOutput, err := parseStopwatchShowOutput(stdout)
+	return parsedOutput, err //need to return err, as it could be set in the defer
+}
+
+// parseStopwatchShowOutput returns the number of total samples for each poll loop
+func parseStopwatchShowOutput(output string) (map[string]stopwatchStatistics, error) {
+	result := make(map[string]stopwatchStatistics)
+
+	reStatisticsBlock := regexp.MustCompile(`(?m)^Statistics for '(?P<name>.+)'$(?:\n  .*)+`)
+	reTotalSamples := regexp.MustCompile(`(?m)^  Total samples: (?P<totalSamples>\d*)$`)
+	reMaximum := regexp.MustCompile(`(?m)^  Maximum: (?P<max>\d*) msec$`)
+	reMinumum := regexp.MustCompile(`(?m)^  Minimum: (?P<min>\d*) msec$`)
+	reShortTermAvg := regexp.MustCompile(`(?m)^  Short term average: (?P<shortTermAvg>\d*\.?\d*) msec$`)
+	reLongTermAvg := regexp.MustCompile(`(?m)^  Long term average: (?P<longTermAvg>\d*\.?\d*) msec$`)
+
+	nameIndex := reStatisticsBlock.SubexpIndex("name")
+	totalSamplesIndex := reTotalSamples.SubexpIndex("totalSamples")
+	maxIndex := reMaximum.SubexpIndex("max")
+	minIndex := reMinumum.SubexpIndex("min")
+	shortTermAvgIndex := reShortTermAvg.SubexpIndex("shortTermAvg")
+	longTermAvgIndex := reLongTermAvg.SubexpIndex("longTermAvg")
+
+	for _, match := range reStatisticsBlock.FindAllStringSubmatch(output, -1) {
+		sws := stopwatchStatistics{}
+
+		if totalSamplesMatch := reTotalSamples.FindStringSubmatch(match[0]); totalSamplesMatch != nil {
+			sws.totalSamples = totalSamplesMatch[totalSamplesIndex]
+		}
+		if minMatch := reMinumum.FindStringSubmatch(match[0]); minMatch != nil {
+			sws.min = minMatch[minIndex]
+		}
+		if maxMatch := reMaximum.FindStringSubmatch(match[0]); maxMatch != nil {
+			sws.max = maxMatch[maxIndex]
+		}
+		if shortTermAvgMatch := reShortTermAvg.FindStringSubmatch(match[0]); shortTermAvgMatch != nil {
+			sws.shortTermAvg = shortTermAvgMatch[shortTermAvgIndex]
+		}
+		if longTermAvgMatch := reLongTermAvg.FindStringSubmatch(match[0]); longTermAvgMatch != nil {
+			sws.longTermAvg = longTermAvgMatch[longTermAvgIndex]
+		}
+
+		metricName := match[nameIndex]
+		result[metricName] = sws
+	}
+
+	return result, nil
+}
+
+// stopwatchShowMetricsUpdater updates the metric by obtaining the stopwatch/show
+// metrics for the specified component.
+func stopwatchShowMetricsUpdater(component string) {
+	for {
+		time.Sleep(30 * time.Second)
+
+		stopwatchShowOutputMap, err := getStopwatchShowOutputMap(component)
```

Comment:
should we check if theres no output from this? i.e. no matching occurred and its an empty map. We should log this and not continue below. WDYT?

Reply:
I didn't check for it, because in that case the values would be set to 0 anyhow in the `parseMetricToFloat` function. Added now a warning and continuing, if the map is empty. WDYT about `continue`ing? Or should we set the metrics in this case to 0, as via the `parseMetricToFloat` function?

Reply:
Looks good to me now after your update.